src/lib.rs

```diff
@@ -17,6 +17,8 @@
 #![doc(html_root_url = "https://docs.rs/num-traits/0.2")]
 #![deny(unconditional_recursion)]
 #![no_std]
+#![cfg_attr(feature = "f16", feature(f16))]
+#![cfg_attr(feature = "f128", feature(f128))]
```

Comment:
I'm not keen on maintaining for unstable features in this crate, so we're probably going to be blocked here until the language and standard library stabilize these new floats.

At that point, I think we'll probably want to have detection in the build script to enable it automatically, and Cargo features can force that on -- like we used to for `has_i128`.